### PR TITLE
fix: read first locale from skill.json if not provided by user

### PR DESCRIPTION
**Pull request by kakhaUrigashvili**
addressing issue from https://github.com/alexa/ask-cli/issues/220

### CODE REVIEW

File: lib/commands/dialog/index.js [hunk 1]
```diff
@@ -96,7 +97,12 @@ class DialogCommand extends AbstractCommand {
                     throw `Failed to obtain skill-id from project resource file ${CONSTANTS.FILE_PATH.ASK_RESOURCES_JSON_CONFIG}`;
                 }
             }
-            locale = locale || process.env.ASK_DEFAULT_DEVICE_LOCALE;
+            const skillPackageSrc = ResourcesConfig.getInstance().getSkillMetaSrc(profile);
+            const manifestPath = path.join(skillPackageSrc, CONSTANTS.FILE_PATH.SKILL_PACKAGE.MANIFEST);
+            new Manifest(manifestPath);
+            const firstLocaleFromManifest = Object.keys(Manifest.getInstance().getPublishingLocales())[0];
```
Comment (kakhaUrigashvili) [inline lib/commands/dialog/index.js:103]: using first locale from skill manifest if not provided

File: lib/commands/dialog/index.js [hunk 2]
```diff
@@ -96,7 +97,12 @@ class DialogCommand extends AbstractCommand {
                     throw `Failed to obtain skill-id from project resource file ${CONSTANTS.FILE_PATH.ASK_RESOURCES_JSON_CONFIG}`;
                 }
             }
-            locale = locale || process.env.ASK_DEFAULT_DEVICE_LOCALE;
+            const skillPackageSrc = ResourcesConfig.getInstance().getSkillMetaSrc(profile);
```
Comment (RonWang) [inline lib/commands/dialog/index.js:100]: If skillId is from the user input, then ask-resources.json might not exist... That's for the use case when users don't have a project locally but doesn't bother them talking to the skill.

This means if skillId is provided by users, then locale value must be provided.

File: lib/commands/dialog/index.js [hunk 3]
```diff
@@ -96,7 +97,12 @@ class DialogCommand extends AbstractCommand {
...
+            const manifestPath = path.join(skillPackageSrc, CONSTANTS.FILE_PATH.SKILL_PACKAGE.MANIFEST);
+            new Manifest(manifestPath);
+            const firstLocaleFromManifest = Object.keys(Manifest.getInstance().getPublishingLocales())[0];
+
+            locale = locale || process.env.ASK_DEFAULT_DEVICE_LOCALE || firstLocaleFromManifest;
```
Comment (RonWang) [inline lib/commands/dialog/index.js:105]: Can we display a message here if locale is the firstLocalFromManifest?

File: lib/commands/dialog/index.js [hunk 4]
```diff
@@ -95,8 +97,16 @@ class DialogCommand extends AbstractCommand {
                 if (!stringUtils.isNonBlankString(skillId)) {
                     throw `Failed to obtain skill-id from project resource file ${CONSTANTS.FILE_PATH.ASK_RESOURCES_JSON_CONFIG}`;
                 }
+                const skillPackageSrc = ResourcesConfig.getInstance().getSkillMetaSrc(profile);
+                const manifestPath = path.join(skillPackageSrc, CONSTANTS.FILE_PATH.SKILL_PACKAGE.MANIFEST);
+                new Manifest(manifestPath);
+                [firstLocaleFromManifest] = Object.keys(Manifest.getInstance().getPublishingLocales());
```
Comment (RonWang) [inline lib/commands/dialog/index.js:103]: ah this is a [niceTrick] lol

**Pull request merged**